src/app/modules/shared_modules/connector/controller.nim

```diff
@@ -26,10 +26,16 @@ QtObject:
   proc delete*(self: Controller) =
     self.QObject.delete
 
-  proc dappRequestsToConnect*(self: Controller, requestId: string, payload: string) {.signal.}
-  proc dappValidatesTransaction*(self: Controller, requestId: string, payload: string) {.signal.}
-  proc dappGrantDAppPermission*(self: Controller, payload: string) {.signal.}
-  proc dappRevokeDAppPermission*(self: Controller, payload: string) {.signal.}
+  proc connectRequested*(self: Controller, requestId: string, payload: string) {.signal.}
```

Comment:
Added request-response pair for the API. This is needed in the UI because it's reacting only on the response. From this point on the nim requests can be done asynchronous.